sdk/eventgrid/azure-eventgrid/azure/eventgrid/_operations/_patch.py

```diff
@@ -123,17 +202,26 @@ def publish_cloud_events(
         :rtype: None
         :raises ~azure.core.exceptions.HttpResponseError:
         """
+        
+        # Check that the body is a CloudEvent or list of CloudEvents even if dict
+        if isinstance(body, dict) or (isinstance(body, list) and isinstance(body[0], dict)):
+            try:
+                if isinstance(body, list):
+                    body = [CloudEvent.from_dict(event) for event in body]
+                else:
+                    body = CloudEvent.from_dict(body)
+            except AttributeError:
+                raise TypeError("Incorrect type for body. Expected CloudEvent or list of CloudEvents.")
```

Comment:
non-blocking: Maybe something like the following to indicate that dicts, if passed must be in CloudEvent format:
```suggestion
                raise TypeError("Incorrect type for body. Expected CloudEvent, list of CloudEvents, dict, or list of dicts. If dict passed, must follow the CloudEvent format.")
```